### PR TITLE
Rebase patches for fixing haiku build.

### PR DESCRIPTION
**Pull request by pulkomandy**
As requested in #18310.

### CODE REVIEW

**Comment by pulkomandy** [comment]
I have a problem: clang-format complains that the includes are not ordered alphabetically. This is on purpose, kernel/image.h must be included before other system include files, because they include image.h and end up getting the one from godot instead of the one from system directories.

**Comment by Faless** [comment]
@pulkomandy 
Put this wherever you want, it will disable clang-formatting between the comments on/off. :)
```
// This must always be included first, disabling clang format
/* clang-format off */
#include <kernel/image.h> // needed for image_id
/* clang-format on */
```

**Comment by fire** [comment]
Alternatively leave a space between the entries.

**Comment by pulkomandy** [comment]
There was a build failure but it looks unrelated. What happened?

**Comment by akien-mga** [comment]
Thanks a lot! Looks good to me.

**Comment by pulkomandy** [comment]
Upstream PRs submitted:
https://github.com/Dav1dde/glad/pull/170
https://github.com/Thekla/thekla_atlas/pull/22

File: thirdparty/glad/glad.c [hunk 1]
```diff
@@ -77,7 +77,7 @@ void close_gl(void) {
 #include <dlfcn.h>
 static void* libGL;
 
-#ifndef __APPLE__
+#if !defined(__APPLE__) && !defined(__HAIKU__)
```
[the remark this answers is not in the document]
Comment (Dav1dde) [reply thirdparty/glad/glad.c]: Will be merged upstream, I'll get to it tomorrow (maintainer of glad here).

**Pull request merged**